thirdeye/thirdeye-frontend/app/pods/components/rootcause-trend/component.js

```diff
@@ -0,0 +1,389 @@
+import Component from '@ember/component';
+import {
+  toCurrentUrn,
+  toBaselineUrn,
+  toOffsetUrn,
+  hasPrefix,
+  filterPrefix,
+  toMetricLabel,
+  isInverse,
+  toColorDirection,
+  makeSortable
+} from 'thirdeye-frontend/utils/rca-utils';
+import {
+  humanizeChange,
+  humanizeFloat,
+  humanizeScore
+} from 'thirdeye-frontend/utils/utils';
+import moment from 'moment';
+import _ from 'lodash';
+
+const ROOTCAUSE_TREND_MAX_COLUMNS = 12;
+
+export default Component.extend({
+  classNames: ['rootcause-metrics'],
+
+  //
+  // external properties
+  //
+
+  /**
+   * Entities cache
+   * @type {object}
+   */
+  entities: null,
+
+  /**
+   * Metric aggregates
+   * @type {object}
+   */
+  timeseries: null,
+
+  /**
+   * Investigation context
+   * @type {object}
+   */
+  context: null,
+
+  /**
+   * User-selected urns
+   * @type {Set}
+   */
+  selectedUrns: null,
+
+  /**
+   * Callback on metric selection
+   * @type {function}
+   */
+  onSelection: null, // function (Set, state)
+
+  /**
+   * loading status for component
+   * @type {boolean}
+   */
+  isLoading: false,
+
+  //
+  // internal properties
+  //
+
+  /**
+   * Display flag for table. Hack for ember models table only re-rendering columns on init()
+   * @type {bool}
+   */
+  showTable: true,
+
+  /**
+   * Hack for ember models table only re-rendering columns on init()
+   */
+  _columnObserver: observer('columns', function () {
+    this.set('showTable', false);
+    later(this, () => this.set('showTable', true), 0);
+  }),
+
+  /**
+   * Start time for time table, overridden by drop down
+   * @type {int}
+   */
+  startTime: computed('context', function () {
+    return this.get('context').analysisRange[0];
+  }),
+
+  /**
+   * Start time formatted
+   * @type {string}
+   */
+  startTimeFormatted: computed('startTime', function () {
+    return this._formatTime(this.get('startTime'));
+  }),
+
+  /**
+   * Start time options for dropdown
+   */
+  startTimeOptions: computed('availableBuckets', function () {
+    const { availableBuckets } = this.getProperties('availableBuckets');
+
+    const options = [];
+    for (let i = 0; i < availableBuckets.length; i += ROOTCAUSE_TREND_MAX_COLUMNS) {
+      options.push(this._formatTime(moment(availableBuckets[i])));
+    }
+
+    return options;
+  }),
+
+  /**
+   * Reverse lookup mapping for start time options for dropdown
+   */
+  startTimeOptionsMapping: computed('availableBuckets', function () {
+    const { availableBuckets } = this.getProperties('availableBuckets');
+
+    const options = {};
+    for (let i = 0; i < availableBuckets.length; i += ROOTCAUSE_TREND_MAX_COLUMNS) {
+      options[this._formatTime(moment(availableBuckets[i]))] = availableBuckets[i];
+    }
+
+    return options;
+  }),
+
+  /**
+   * Possible time buckets from analysis range
+   * @type {int[]}
+   */
+  availableBuckets: computed('context', function () {
+    const { context } = this.getProperties('context');
```

Comment:
Single retrieval of a prop can use `this.get()` instead.

`const context  = this.get('context');`

and all other places in this component code. Thanks.

Reply:
@apucher Yup that's fine, the same way we use both `set` and `setProperties`. The only difference is that `setProperties` is more performant (each `this.set()` would notify a property change, and cause `cp`s and `observer`s to recalculate), but `getProperties` and `get` are essentially doing the same thing.